### ISSUE TITLE
Support Component Sub Types

### ISSUE THREAD

**Issue by korgon** [searchspring/snapfu#127]
With the addition of several new "bundle" recommendation components to the library, we will want to refactor Snapfu (and the library) to support selection of these sub types when using `snapfu recs init`. We may also want to support this grouping for all component groups for future collections.

The groupings just allow for an additional level of organization within the Snapfu CLI and the library. Snapfu should recognize the hierarchy and present the user with an additional question asking for the specific component to use within the group selected. If there is no additional hierarchy (grouping) the CLI should work as it does currently (see badge example below).

Current structure:
```
components: {
  recommendation: {
    default,
    bundle,
    email,
    quickAdd,
    vertical,
    list
  }
  badge: {
    default
  }
}
```

Proposed changes:
```
components: {
  recommendation: {
    standard: {
      default
    },
    bundle: {
      default,
      quickAdd,
      vertical,
      list
    },
    email: {
      default
    },
  }
  badge: {
    default
  }
}
```

**Comment by korgon** [comment]
Changing scope slightly... for components it would be nice to have some variables that the init command would inquire the user for (similar to scaffolds):

The library setup would need to ingest (and parse) the variables if they exist, and then ask questions based on that. The answers to those questions would be passed on to the `copy` utility that will find and replace them when copying the files.
https://github.com/searchspring/snapfu/blob/main/src/recs.js#L114-L118 - variables should begin with `snapfu.variables`.

Proposed change to library setup to add the variables:

```
componentCategory[componentTypeFile][componentFile] = {
	path: componentFilePath,
	label: componentFile,
	files: filteredComponentContents,
	variables: undefined | parsedComponentDescriptorYaml
};
```

Suggested loop through custom variables _after_ questions have been answered:

```
customVariables.forEach((n, variable) => {
	variables[`snapfu.variables.${variable.name}`]: customAnswers[n],
});
```


Example component descriptor file (would live in the library for components that need them):

```
# component descriptor file

# custom component variables
variables: [
  {
    name: 'addToCart',
    message: 'Choose an addToCart method:',
    type: 'list',
    choices: ['shopify', 'magento2', 'bigcommerce', 'custom'],
    choicesMap: {
      shopify: "import { addToCart } from '@searchspring/snap-platforms/shopify';",
      magento2: "import { addToCart } from '@searchspring/snap-platforms/magento2';",
      bigcommerce: "import { addToCart } from '@searchspring/snap-platforms/bigcommerce';",
      custom: "const addToCart = (items) => { console.log('') };"
    },
    default: 'custom',
  }
]
```

**Comment by korgon** [comment]
Released in v1.8.0.
